tests/test_node.py

```diff
@@ -67,3 +67,46 @@ def test_nodejs_attach_maps_from_container(
     )
     collapsed = parse_one_collapsed(collapsed_text)
     assert_collapsed(collapsed)
+
+@pytest.mark.parametrize("in_container", [True])
+@pytest.mark.parametrize(
+    "application_image_tag", 
+    [
+        "10-alpine",
+        "10-slim",
+        "11-alpine",
+        "11-slim",
+        "12-alpine",
+        "12-slim",
+        "13-alpine",
+        "13-slim",
+        "14-alpine",
+        "14-slim",
+        "15-alpine",
+        "15-slim",
+        "16-alpine",
+        "16-slim",
+    ],
+)
+@pytest.mark.parametrize("profiler_type", ["attach-maps"])
+@pytest.mark.parametrize("runtime", ["nodejs"])
+def test_nodejs_matrix(
+    docker_client: DockerClient,
+    application_pid: int,
+    assert_collapsed: AssertInCollapsed,
+    gprofiler_docker_image: Image,
+    output_directory: Path,
+    output_collapsed: Path,
+    runtime_specific_args: List[str],
+    profiler_flags: List[str],
+    application_image_tag: str,
+    ) -> None:
+    _ = application_pid
+    node_version = application_image_tag.split("-")[0]
+    if node_version in ["14", "15", "16"]:
+        pytest.xfail("This test fails with these nodejs versions")
+    collapsed_text = run_gprofiler_in_container_for_one_session(
```

Comment:
For simpler & faster tests - we don't need a full run of the gProfiler container here, let's initialize `SystemProfiler` and use it here, i.e like `test_nodejs_attach_maps`.